src/Generator/Passes/GetterSetterToPropertyPass.cs

```diff
@@ -177,6 +177,8 @@ private static void GenerateProperty(string name, DeclarationContext context, Me
                 }
                 property.GetMethod = getter;
                 property.SetMethod = setter;
+                if (setter != null && !setter.ReturnType.Type.IsPrimitiveType(PrimitiveType.Void))
+                    setter.ReturnType = new QualifiedType(new BuiltinType(PrimitiveType.Void));
```

Comment:
Wouldn't this negatively affect the internal return type and therefore the P/Invocation? Does it work with an indirect return param?